File: Python/202. Happy Number.py
class Solution:
    def isHappy(self, n: int) -> bool:
        set = {}
        x = n
        while set.get(x) is None:
            if x == 1:
                return True
            set[x] = True
            sum = 0
            while x != 0:
                tmp = x % 10
                sum += tmp * tmp
                x = x // 10
            x = sum
        return False

File: C++/202. Happy Number.cpp
#include <iostream>
#include <unordered_set>

class Solution {
public:
    bool isHappy(int n) {
        std::unordered_set<int> set;
        for (int x = n; set.find(x) == set.end(); ) {
            if (x == 1)
                return true;
            set.insert(x);
            int sum = 0;
            while (x != 0) {
                int tmp = x % 10;
                sum += tmp * tmp;
                x = x / 10;
            }
            x = sum;
        }
        return false;
    }
};
